crates/llm_gateway/src/stream_context.rs

```diff
@@ -527,12 +542,15 @@ impl StreamContext {
                         }
                     }
 
-                    // Add transformed event to response buffer
-                    let bytes: Vec<u8> = transformed_event.into();
-                    response_buffer.extend_from_slice(&bytes);
+                    // Add transformed event to buffer (buffer may inject lifecycle events)
+                    self.sse_buffer
+                        .as_mut()
+                        .unwrap()
+                        .add_transformed_event(transformed_event);
                 }
 
-                Ok(response_buffer)
+                // Get accumulated bytes from buffer and return
+                Ok(self.sse_buffer.as_mut().unwrap().into_bytes())
             }
```

Comment:
`.unwrap()` could cause panic. I think you can work off from sse_buffer var you create from line 484 and then use that var at like 545 and 552.